### PR TITLE
FTP-1381:form-fields

### PR DESCRIPTION
**Pull request by HamidAli111**
Screenshot 2022-10-03 at 1 58 12 PM
Screenshot 2022-10-03 at 1 58 19 PM
Screenshot 2022-10-03 at 1 58 42 PM
Screenshot 2022-10-03 at 1 58 48 PM
Screenshot 2022-10-03 at 1 58 52 PM
Screenshot 2022-10-03 at 1 58 59 PM
Screenshot 2022-10-06 at 4 56 31 PM

Screenshot 2022-10-03 at 2 03 01 PM

### CODE REVIEW

File: components/button/Tag.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { render, screen } from '@testing-library/react';
+import Tag from './Tag';
```
Comment (skippednote) [inline components/button/Tag.spec.tsx:2]: Let's name it time

File: @types/Searchbox.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+type searchboxType = {
```
Comment (skippednote) [inline @types/Searchbox.ts:1]: Let's aggregate all form-related types in a `form.types.ts` file

File: components/inputField/InputField.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import inputType from '../../@types/input.types';
+
+function InputField({ label, variant, placeholder, icon }: inputType) {
+  let inpType = '';
```
Comment (skippednote) [inline components/inputField/InputField.tsx:5]: let's write the whole thing

File: components/inputField/InputField.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  }
+  if (variant == 'date') {
+    inpType = 'date';
+  }
+
+  if (icon == 'left') {
+    return (
+      <>
+        <div>
+          <h1 className={extraClasses}>{label}</h1>
```
Comment (skippednote) [inline components/inputField/InputField.tsx:31]: This should be a label

File: components/inputField/InputField.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import inputType from '../../@types/input.types';
+
+function InputField({ label, variant, placeholder, icon }: inputType) {
+  let inpType = '';
+  let iconLeft = '';
+  let iconRight = '';
+  const extraClasses = 'pl-[61px] absolute text-sm mt-1.5';
```
Comment (skippednote) [inline components/inputField/InputField.tsx:8]: This should be renamed to `labelClasses`

**Comment by skippednote** [comment]
Replace `.spec.ts` with `.test.ts`
Replace `.png` with Icon component

**Review by skippednote — CHANGES_REQUESTED** [review]

File: @types/Searchbox.ts (hunk 2 above)
Comment (HamidAli111) [reply @types/Searchbox.ts:1]: As there were some styling issues is the reason we have to keep it as separate from the inputfields

File: components/button/Tag.spec.tsx (hunk 1 above)
Comment (HamidAli111) [reply components/button/Tag.spec.tsx]: Made the required changes

File: components/inputField/InputField.tsx (hunk 3 above)
Comment (HamidAli111) [reply components/inputField/InputField.tsx]: Made the required chnages

File: components/inputField/InputField.tsx (hunk 4 above)
Comment (HamidAli111) [reply components/inputField/InputField.tsx]: Made the required chnages

File: components/inputField/InputField.tsx (hunk 5 above)
Comment (HamidAli111) [reply components/inputField/InputField.tsx:8]: Made the required chnages

File: pages/testing.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
```
Comment (skippednote) [inline pages/testing.tsx:1]: Please remove

File: pages/testing.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          <Time start="10PM" end="12AM" variant="inactive" />
+          <Time start="10PM" end="12AM" variant="inactive" />
+          <Time start="10PM" end="12AM" variant="inactive" />
+        </div>
+      </div>
+    </>
+  );
+}
+
+export default Testing;
```
Comment (skippednote) [inline pages/testing.tsx:44]: Remove all the png icons.

File: components/inputField/Searchbox.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    <>
+      {variant == 'active' ? (
+        <label className="text-md text-red-primary pl-60 absolute mt-7 leading-xs">
+          {label}
+        </label>
+      ) : (
+        ''
+      )}
+      <div
+        data-testid="box"
```
Comment (skippednote) [inline components/inputField/Searchbox.tsx:17]: let's use a more verbose name instead of `box`

File: components/inputField/Searchbox.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+const Template: ComponentStory<typeof Searchbox> = (args) => (
+  <Searchbox {...args} />
+);
+
+export const Primary = Template.bind({});
+Primary.args = {
+  variant: 'active',
+  label: 'Search',
+  placeholder: 'Search Anything',
+  icon: '/icons/search_24px.png',
```
Comment (skippednote) [inline components/inputField/Searchbox.stories.tsx:22]: Can we remove this?

**Review by skippednote — CHANGES_REQUESTED** [review]

File: pages/testing.tsx (hunk 7 above)
Comment (HamidAli111) [reply pages/testing.tsx]: Made the required chnages

File: pages/testing.tsx (hunk 6 above)
Comment (HamidAli111) [reply pages/testing.tsx]: Made the required chnages

File: components/inputField/Searchbox.tsx (hunk 8 above)
Comment (HamidAli111) [reply components/inputField/Searchbox.tsx]: Made the required chnages

File: components/inputField/Searchbox.stories.tsx (hunk 9 above)
Comment (HamidAli111) [reply components/inputField/Searchbox.stories.tsx]: Made the required chnages

File: components/time/Time.test.tsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import { render, screen } from '@testing-library/react';
+import Time from '../time/Time';
+
+describe('Test Time component', () => {
+  it('renders the component and checks for the specific class', () => {
+    const StartTime = '10PM';
+    const EndTime = '10AM';
```
Comment (skippednote) [inline components/time/Time.test.tsx]: const should either be upper case or camelCase, not capitalcase

File: components/time/Time.test.tsx [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    render(<Time start={StartTime} end={EndTime} variant="active" />);
+    const time = screen.getByRole('button');
+    expect(time).toHaveClass('text-red-primary');
+  });
+  it('renders the component and checks for the specific class', () => {
+    const StartTime = '10PM';
+    const EndTime = '10AM';
```
Comment (skippednote) [inline components/time/Time.test.tsx]: const should either be upper case or camelCase, not capitalcase

**Review by skippednote — APPROVED** [review]

**Pull request merged**